src/main/java/duke/core/Duke.java

```diff
@@ -0,0 +1,81 @@
+package duke.core;
+
+import java.util.Scanner;
+
+import duke.command.Commandable;
+import duke.exception.DukeException;
+
+/**
+ * Encapsulates the Main Program where Duke is run.
+ *
+ * @author Clifford
+ */
+public class Duke {
+    private static final String TASKS_FILE_PATH = "bin/data/tasks.txt";
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    /**
+     * Initialises Duke for whole application to run
+     */
+    public Duke() {
+        ui = new Ui();
+        try {
+            storage = new Storage(TASKS_FILE_PATH);
+            taskList = new TaskList(storage);
+            taskList.retrieveTasks();
+        } catch (DukeException e) {
+            ui.formatDisplay(e.getMessage());
+        }
+    }
+
+    /**
+     * Sends a greeting message when the user first starts up Duke
+     * @return the greeting message when user starts up Duke
+     */
+    public String greetUser() {
+        return "Hello! I'm Duke!\nHow may I be of service to you?";
```

Comment:
I think that this method would fit well in Ui.java instead of having it here, would also be easier for you to change the text in the future since all the UI messages are grouped together :)

Reply:
That's a good point. Thank you for raising this.